src/controller/controller.cpp

```diff
@@ -213,51 +185,32 @@ void Controller::runCommandHandler(const std::vector<ElectronicID::ptr>& availab
 
 void Controller::onCertificatesLoaded()
 {
-    CardEventMonitorThread* cardEventMonitorThread =
-        new CardEventMonitorThread(this, std::string(commandType()));
-    saveChildThreadPtrAndConnectFailureFinish(cardEventMonitorThread);
-    cardEventMonitorThreadKey = uintptr_t(cardEventMonitorThread);
+    auto* cardEventMonitorThread = new CardEventMonitorThread(this, commandType());
+    connect(cardEventMonitorThread, &ControllerChildThread::failure, this,
+            &Controller::onCriticalFailure);
+    connect(this, &Controller::stopCardEventMonitorThread, cardEventMonitorThread,
+            &WaitForCardThread::requestInterruption);
```

Comment:
As `cardEventMonitorThread` is `CardEventMonitorThread*`, connect to `&CardEventMonitorThread::requestInterruption`:
```suggestion
            &CardEventMonitorThread::requestInterruption);
```